### PR TITLE
Index repair improvements

### PR DESCRIPTION
**Pull request by gguttikonda**
* Changed the workblock size to be storage.page-size ( this makes the pagination code cleaner ) and also make workblock configurable.
* Removed QUEUE_SIZE from Scanner, it now follows storage.page-size
* ProcessQueue size would be numWorkers * workBlockSize * numDataPullers
* Add a helper method in KeyIterator. This helps removing the while
condition with retry/fixed time-out to Backend based exhaustion. This is
very helpful for storage backends like CQL.
* Add info logs for every job iteration to see the index update metrics.
* Handle the `doc-updates` count accurately matching with actual
updates.
* Change the CQLKeyColumnValueStore to use 'pageStateUnsafe' to avoid
very expensive md5 compute.

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import org.janusgraph.diskstorage.indexing.IndexInformation;
+import org.janusgraph.graphdb.database.serialize.Serializer;
+import org.janusgraph.graphdb.internal.ElementCategory;
+import org.janusgraph.graphdb.internal.ElementLifeCycle;
+import org.janusgraph.graphdb.transaction.StandardJanusGraphTx;
+import org.janusgraph.graphdb.types.IndexField;
+import org.janusgraph.graphdb.types.MixedIndexType;
+import org.janusgraph.graphdb.types.ParameterIndexField;
+import org.janusgraph.graphdb.vertices.StandardVertex;
+import org.junit.BeforeClass;
```
Comment (FlorianHockmann) [inline janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.junit.BeforeClass'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4608826)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 6
           

Complexity increasing per file
==============================
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  1
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
         

Clones added
============
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/janusgraph/janusgraph/pullRequest?prid=4608826&bid=15462723)

File: janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.database;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
```
Comment (FlorianHockmann) [inline janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert.fail'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4608826)

File: janusgraph-cql/src/test/java/org/janusgraph/diskstorage/cql/CQLResultSetKeyIteratorTest.java [hunk 3]
```diff
@@ -14,10 +14,9 @@
 
 package org.janusgraph.diskstorage.cql;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
+import static org.junit.Assert.*;
```
Comment (FlorianHockmann) [inline janusgraph-cql/src/test/java/org/janusgraph/diskstorage/cql/CQLResultSetKeyIteratorTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4608826)

File: janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import org.janusgraph.core.schema.Parameter;
+import org.janusgraph.core.schema.SchemaStatus;
+import org.janusgraph.diskstorage.configuration.Configuration;
+import org.janusgraph.diskstorage.indexing.IndexEntry;
+import org.janusgraph.diskstorage.indexing.IndexInformation;
+import org.janusgraph.graphdb.database.serialize.Serializer;
+import org.janusgraph.graphdb.internal.ElementCategory;
+import org.janusgraph.graphdb.internal.ElementLifeCycle;
+import org.janusgraph.graphdb.transaction.StandardJanusGraphTx;
+import org.janusgraph.graphdb.types.IndexField;
```
Comment (FlorianHockmann) [inline janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.janusgraph.graphdb.types.IndexField'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4608826)

File: janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+// limitations under the License.
+
+package org.janusgraph.graphdb.database;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.when;
```
Comment (FlorianHockmann) [inline janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.mockito.Mockito.when'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4608826)

File: janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.database;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.*;
```
Comment (FlorianHockmann) [inline janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.mockito.Mockito'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4608826)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  1
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
         

Clones added
============
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/janusgraph/janusgraph/pullRequest?prid=4608826&bid=15462723)

**Comment by gguttikonda** [comment]
Will create a new PR with one change and DCO taken care of

**Pull request closed without merging**